apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -157,19 +164,27 @@ export class ApplicationController {
   })
   @ApiOkResponse({ type: ApplicationResponseDto, isArray: true })
   @UseInterceptors(ApplicationSerializer)
+  @Audit<ApplicationResponseDto[]>({
+    resources: (apps) => apps.map((app) => app.id),
+  })
   async findAll(
-    @CurrentRestUser() user: User,
+    @Param('nationalId') nationalId: string,
+    @CurrentUser() user: User,
     @Query('typeId') typeId?: string,
     @Query('status') status?: string,
   ): Promise<ApplicationResponseDto[]> {
+    if (nationalId !== user.nationalId) {
+      throw new UnauthorizedException()
+    }
```

Comment:
Suggestion for improvement: this to be extracted as part of a guard ... somehow ;)

Reply:
Future improvement I meant ;)